compiler/src/iree/compiler/Codegen/Dialect/GPU/TargetUtils/ConfigUtils.cpp

```diff
@@ -468,4 +468,74 @@ LogicalResult setTileAndFuseLoweringConfig(IREE::GPU::TargetAttr target,
       workgroupSize, subgroupSize, DictionaryAttr());
 }
 
+//===----------------------------------------------------------------------===//
+// Lowering Config Attributes
+//===----------------------------------------------------------------------===//
+
+GPUPipelineOptions
+getPipelineOptions(FunctionOpInterface funcOp,
+                   IREE::Codegen::TranslationInfoAttr translationInfo) {
+  GPUPipelineOptions pipelineOptions = {};
+  auto targetAttr = IREE::HAL::ExecutableTargetAttr::lookup(funcOp);
+
+  if (DictionaryAttr config = translationInfo.getConfiguration()) {
+
+    std::optional<NamedAttribute> pipelineOptionsListAttr =
+        config.getNamed(GPUPipelineOptionsArrayAttr::getDictKeyName());
+    if (!pipelineOptionsListAttr.has_value()) {
+      return pipelineOptions;
+    }
+    ArrayRef<GPUPipelineOptionAttr> pipelineOptionsList =
+        cast<GPUPipelineOptionsArrayAttr>(pipelineOptionsListAttr->getValue())
+            .getValue();
+    for (GPUPipelineOptionAttr pipelineAttr : pipelineOptionsList) {
+      switch (pipelineAttr.getValue()) {
+      case GPUPipelineOption::NoReduceSharedMemoryBankConflicts:
+        pipelineOptions.enableReduceSharedMemoryBankConflicts = false;
+        continue;
+      case GPUPipelineOption::PrefetchSharedMemory:
+        pipelineOptions.prefetchSharedMemory = true;
+        continue;
+      case GPUPipelineOption::ReorderWorkgroupsNone:
+      case GPUPipelineOption::ReorderWorkgroupsSwizzle:
+      case GPUPipelineOption::ReorderWorkgroupsTranspose:
+        pipelineOptions.reorderStrategy =
+            static_cast<ReorderWorkgroupsStrategy>(pipelineAttr.getValue());
```

Comment:
doesn't tablegen generate some helper to get all enum values that we could loop over? alternatively, could we use `llvm::enum_seq`?